debian/patches/wazo_mixmonitor_events

```diff
@@ -11,51 +11,81 @@ Index: asterisk-18.2.0/apps/app_mixmonitor.c
  #include "asterisk/callerid.h"
  #include "asterisk/mod_format.h"
  #include "asterisk/linkedlists.h"
-@@ -1077,6 +1079,7 @@ static int mixmonitor_exec(struct ast_ch
- 	struct ast_flags flags = { 0 };
- 	char *recipients = NULL;
- 	char *parse;
+@@ -58,6 +60,7 @@
+ #include "asterisk/mixmonitor.h"
+ #include "asterisk/format_cache.h"
+ #include "asterisk/beep.h"
++#include "asterisk/cel.h"
+ 
+ /*** DOCUMENTATION
+ 	<application name="MixMonitor" language="en_US">
+@@ -893,6 +896,8 @@ static int launch_monitor_thread(struct
+ 	struct mixmonitor *mixmonitor;
+ 	char postprocess2[1024] = "";
+ 	char *datastore_id = NULL;
 +	RAII_VAR(struct stasis_message *, message, NULL, ao2_cleanup);
- 	AST_DECLARE_APP_ARGS(args,
- 		AST_APP_ARG(filename);
- 		AST_APP_ARG(options);
-@@ -1197,6 +1200,13 @@ static int mixmonitor_exec(struct ast_ch
- 		ast_module_unref(ast_module_info->self);
++	RAII_VAR(struct ast_json *, blob, NULL, ast_json_unref);
+ 
+ 	postprocess2[0] = 0;
+ 	/* If a post process system command is given attach it to the structure */
+@@ -1014,6 +1019,15 @@ static int launch_monitor_thread(struct
+ 		mixmonitor_free(mixmonitor);
+ 		return -1;
  	}
++	blob = ast_json_pack("{s: s, s: s}",
++		"filename", filename,
++		"mixmonitor_id", datastore_id);
++	message = ast_channel_blob_create_from_cache(ast_channel_uniqueid(chan),
++												ast_channel_mixmonitor_start_type(),
++												blob);
++	if (message) {
++		stasis_publish(ast_channel_topic(chan), message);
++	}
  
-+  message = ast_channel_blob_create_from_cache(ast_channel_uniqueid(chan),
-+                                               ast_channel_mixmonitor_start_type(),
-+                                               NULL);
-+  if (message) {
-+    stasis_publish(ast_channel_topic(chan), message);
-+  }
-+
- 	return 0;
- }
+ 	ast_free(datastore_id);
  
-@@ -1206,6 +1216,7 @@ static int stop_mixmonitor_full(struct a
+@@ -1206,6 +1220,9 @@ static int stop_mixmonitor_full(struct a
  	char *parse = "";
  	struct mixmonitor_ds *mixmonitor_ds;
  	const char *beep_id = NULL;
++	const char *datastore_id = NULL;
 +	RAII_VAR(struct stasis_message *, message, NULL, ao2_cleanup);
++	RAII_VAR(struct ast_json *, blob, NULL, ast_json_unref);
  
  	AST_DECLARE_APP_ARGS(args,
  		AST_APP_ARG(mixmonid);
-@@ -1263,6 +1274,13 @@ static int stop_mixmonitor_full(struct a
+@@ -1229,6 +1246,12 @@ static int stop_mixmonitor_full(struct a
+ 
+ 	ast_mutex_lock(&mixmonitor_ds->lock);
+ 
++	if (ast_asprintf(&datastore_id, "%p", mixmonitor_ds) == -1) {
```

Comment:
What do you think of using `args.mixmonid` instead?

Reply:
args.mixmonid is only valid when it has been specified on the command line

Reply:
OK, thanks!